### ISSUE TITLE
Issue with Marshmallow Module Not Being Found

### ISSUE THREAD

**Issue by tymas1011** [superconductive/great_expectations_lambda#1]
Hi there,

First off, thank you for this guide for the Lambda set up! It was relatively easy to follow along and figure out :)

I have a persistent issue where I try to hit the api endpoint and get an internal server error. Digging through Cloudwatch I am posed with the following:

```
  | 2020-07-15T21:48:05.896-04:00 | deps loaded to tmp
-- | -- | --
  | 2020-07-15T21:48:05.938-04:00 | api - ERROR - Caught exception for <function index at 0x7f9916c773b0>
  | 2020-07-15T21:48:05.938-04:00 | Traceback (most recent call last):
  | 2020-07-15T21:48:05.938-04:00 | File "/var/task/chalice/app.py", line 1111, in _get_view_function_response
  | 2020-07-15T21:48:05.938-04:00 | response = view_function(**function_args)
  | 2020-07-15T21:48:05.938-04:00 | File "/var/task/app.py", line 16, in index
  | 2020-07-15T21:48:05.938-04:00 | import great_expectations
  | 2020-07-15T21:48:05.938-04:00 | File "/opt/python/lib/python3.7/site-packages/great_expectations/__init__.py", line 7, in <module>
  | 2020-07-15T21:48:05.938-04:00 | from great_expectations.data_context import DataContext
  | 2020-07-15T21:48:05.938-04:00 | File "/opt/python/lib/python3.7/site-packages/great_expectations/data_context/__init__.py", line 3, in <module>
  | 2020-07-15T21:48:05.938-04:00 | from .data_context import BaseDataContext, DataContext, ExplorerDataContext
  | 2020-07-15T21:48:05.938-04:00 | File "/opt/python/lib/python3.7/site-packages/great_expectations/data_context/data_context.py", line 17, in <module>
  | 2020-07-15T21:48:05.938-04:00 | from marshmallow import ValidationError
  | 2020-07-15T21:48:05.938-04:00 | ModuleNotFoundError: No module named 'marshmallow'
```

I've tried everything I can think of to attempt to make sure Marshmallow is getting installed, but to no avail. Things I've tried:

- Changing the version in ge-requirements.txt
-- Forcing a version (2.2.0, 3.7.0, etc.)
--No version specified
- Reunning the build_layer.sh script with the above changes
- Attempting to manually add marshmallow to the deps zip (this was more out of desperation, I figured it wouldn't work)

I've verified that Marshmallow is being installed when running the build script as well.

Additionally, I am getting an error when attempting to run the layer build script, I'm not sure if it could be related to the Marshmallow package not being found.

Running on Windows 10. Build script is run using cygwin and the remainder of the guide is completed using PyCharm and it's terminal included in the IDE.

Let me know if there's anything else I can include.

Thanks!

**Comment by tymas1011** [comment]
Found the answer in a discussion on GE website. Copying the answer for anyone with future issues:

> 
beau
Feb 20
tl;dr: use the Amazon Linux docker image (as @joostboonzajerflaes suggests), or an EC2 instance with an Amazon Linux AMI (@beau’s preferred method)

As you note, it is necessary to perform builds under amazon linux. This docs page 1 provides the details on how various lambda runtimes map to the underlying environment.

There are two main ways to easily create an environment that will allow you to transfer builds into a lambda runtime.

You’ve already discovered one, which is to use a suitable docker container. For others’ reference, those docker images can be found here 2 on docker hub.

Because docker is not part of my own dev workflow, I avoid it when possible. Instead, I spin up a suitable Amazon Linux EC2 instance for these builds. If you use the EC2 console to perform the launch, the appropriate Amazon Linux AMIs appear at the very top of their Quick Start list. Here is the complete list of AMI IDs by region and storage type.
